Update build documentation

mkangia opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
Updates documentation to setup HQ locally to submit forms
https://dimagi.slack.com/archives/C02QG63MU/p1633349375299700

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
When we use `runserver` with local IP address to setup access from phone, we also need to update nginx to use the IP address instead of `localhost` for form submissions to work properly.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
Tested that this now works with
1. `./manage.py runserver`
2. setting `BASE_ADDRESS` as `192.168.29.250`
3. replacing `localhost` with `192.168.29.250` in `cchq_local_nginx.conf`

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

mkangia committed:
fixes link

mkangia committed:
updates doc to update nginx conf

snopoke commented on corehq/apps/builds/README.rst:
```diff
@@ -82,7 +82,7 @@ Try accessing ``http://localhost/a/domain`` and see if it works. nginx should
 proxy all requests to localhost to your django server. You should also be able
 
-to access ``http://your_ip_address/a/domain`` from a phone or other device on the
+to access ``http://your_ip_address:8000/a/domain`` from a phone or other device on the
```
you don't need this port if you are using nginx

mkangia replied:
Hey Simon,
When I was setting this up, this did not work without the port.
I think it's because nginx conf by default has setup for localhost, and since we are running runserver with `./manage.py runserver 192.168.1.5:8000`, nginx does not come into play.
It should work without port if localhost is changed with the IP address in the nginx conf, as mentioned later. So I modified this based on if someone is following this documentation in the same order as written.

Feel free to do a quick check locally to confirm or suggest alternate.

snopoke replied:
The section of the doc you are updating is talking specifically about using nginx.

mkangia replied:
Just rechecked this. Its correct that we should not need port. Updated this later to move it after nginx reload though.

RynhardPietersen reviewed (approved):
This is really good information to have.

mkangia committed:
Merge branch 'master' into mk/update-build-documentation

mkangia committed:
fix filename

mkangia committed:
remove incorrect change for proxy_pass

mkangia committed:
fix spelling

mkangia committed:
fix access to be after nginx reload

mkangia committed:
remove port

since django server is running on localhost and nginx server_name is the
IP address, there should be no port added

mkangia commented on corehq/apps/builds/README.rst:
```diff
@@ -91,9 +91,15 @@ Make Commcare use your local IP address
+Additionally, modify ``deployment/nginx/cchq_local_nginx.conf`` to replace localhost with
+your IP address as ``server_name`` and also in ``proxy_pass``.
+For example, set server_name as ``192.168.0.10`` and
+``proxy_pass`` as ``http://192.168.0.10:8000``.
```
note: the change for proxy_pass was removed later.

mkangia commented:
Thanks for the note and the review @RynhardPietersen. I totally lost track of this change until now when I got stuck again with local setup.

mkangia commented:
@snopoke requesting a followup on your last review. Undid some change from last time and made new improvements.

mkangia committed:
add nginx command for macOS

mkangia committed:
reword

mkangia commented:
This is ready to merge now but waiting on confirmation from team if we can ignore the stickler time out on check

mkangia commented:
Also, added equivalent of `nginx -s reload` for mac in instructions.

Thank you @pxwxnvermx for sharing that.
